pylabrobot/machines/machine.py

```diff
@@ -49,6 +49,48 @@ def __init__(
   def setup_finished(self) -> bool:
     return self._setup_finished
 
+  def serialize(self) -> dict:
+    return {**super().serialize(),
+            "backend": self.backend.serialize()}
+
+  @classmethod
+  def deserialize(cls, data: dict):
+    data_copy = data.copy() # copy data because we will be modifying it
+    backend_data = data_copy.pop("backend")
+
+    def find_subclass(
+      class_name: str,
+      cls: Type[MachineBackend] = MachineBackend
+    ) -> Optional[Type[MachineBackend]]:
+      """ Recursively find a MachineBackend with the correct name.
+
+      Args:
+        class_name: The name of the class to find.
+        cls: The class to search in.
+
+      Returns:
+        The class with the given name, or `None` if no such class exists.
+      """
+
+      if cls.__name__ == class_name:
+        return cls
+      for subclass in cls.__subclasses__():
+        subclass_ = find_subclass(class_name=class_name, cls=subclass)
+        if subclass_ is not None:
+          return subclass_
+      return None
```

Comment:
this should be able to use `get_resource_class_from_string(class_name, Machine)`

Reply:
I edited this so it imports the function from from resources

Reply:
realized that this is not going to work due MachineBackend only inheriting from ABC, reverted the changes